src/halmos/sevm.py

```diff
@@ -1352,21 +1380,44 @@ def call(
         if not ret_size >= 0:
             raise ValueError(ret_size)
 
+        if arg_size > 0:
+            arg = wload(ex.st.memory, arg_loc, arg_size)
+        else:
+            arg = None
+
         caller = ex.prank.lookup(ex.this, to)
 
-        orig_code = ex.code.copy()
-        orig_storage = deepcopy(ex.storage)
-        orig_balance = ex.balance
-        orig_log = ex.log.copy()
+        def send_callvalue(condition=None) -> None:
+            if (
+                # no balance update for CALLCODE which transfers to itself
```

Comment:
does that comment describe the code in the if? it's not obvious to me

Reply:
it explains why the msg.value transfer is done only for op == CALL, while CALLCODE also involves msg.value.